aioauth/response_type.py

```diff
@@ -115,12 +115,10 @@ async def create_authorization_response(
             generate_token(48),
         )
         return TokenResponse(
-            expires_in=token.expires_in,
-            refresh_token_expires_in=token.refresh_token_expires_in,
             access_token=token.access_token,
-            refresh_token=token.refresh_token,
```

Comment:
I think I got a bit confused about removing the refresh_token from the response. How will the client get a refresh token if it needs to update the access_token? Auth0 uses the offline_access scope for this: https://auth0.com/docs/secure/tokens/refresh-tokens/get-refresh-tokens

This document says that the response MAY contain a refresh_token: https://www.oauth.com/oauth2-servers/making-authenticated-requests/refreshing-an-access-token/

Reply:
Another solution could be a feature flag. If it's enabled, the refresh token won't be returned in the response. This way, backward compatibility won't be broken, while still adhering to the behavior outlined in the RFC6749 standard.

Reply:
Well, I see your point @aliev. However, this is something very specific of the _Implicit Grant_. Indeed, the RFC 6749 clearly mentions that this flow was not conceived for refresh tokens:

> The implicit grant type is used to obtain access tokens (it does not
   support the issuance of refresh tokens) and is optimized for public
   clients known to operate a particular redirection URI.  These clients
   are typically implemented in a browser using a scripting language
   such as JavaScript.

(source: [RFC 6749, section "4.2 Implicit Grant"](https://datatracker.ietf.org/doc/html/rfc6749#section-4.2))

And also here:

> The authorization server MUST NOT issue a refresh token.
(source: [4.2.2.  Access Token Response](https://datatracker.ietf.org/doc/html/rfc6749#section-4.2.2))

Regarding this Auth0 link you paste before, the blog is very generic explanation about refresh token, so I guess they didn't really mean it in a concrete context like _Implicit Grant_. I think this may be why.

In any case, your proposal here:

> Another solution could be a feature flag. If it's enabled, the refresh token won't be returned in the response. This way, backward compatibility won't be broken, while still adhering to the behavior outlined in the RFC6749 standard.

looks quite fine to me so that it leaves the decision to the implementer to be fully compliant or not!

Thanks again @aliev !!!

Cheers
Jose M. Prieto

PS: by the way, sorry for the long time in the reply, but your know, I was off in my summer break :P

Reply:
Hi @PrieJos! Could you create another PR to fix the issue you mentioned in #97 while I research the refresh token? I would really appreciate your contribution and would be happy to see you on the list of contributors! :)

Reply:
HI @aliev 

Here it goes the PR for issue #97: #103 

Thanks
Jose M. Prieto

Reply:
Hi @PrieJos !

Shall we try to deliver this MR as well?

As I mentioned, the safest way would be to implement a feature flag disabled by default.

You could add it, for example, in the Settings: https://github.com/aliev/aioauth/blob/master/aioauth/config.py#L15

We should also remove the changes related to HTTPHeaderDict from this PR, as they are already in the master branch.

Reply:
Hi @aliev 

Sorry for the long time to reply. I totally oversee the notification. Sorry again.

Yes, that would be great. I will take a look and come out with a suggestion so that you can review it.

Cheers,
Jose M. Prieto

Reply:
Hi @PrieJos 

No worries, thank you for your contribution again!

Reply:
Hi @aliev,

I'll propose to continue the discussion on the original issue #98 instead of this PR since it became now obsolete after I pushed the new PR #108 for the final proposed solution.

Cheers
Jose M. Prieto